### PR TITLE
Provide a list of macros present in the confluence package

### PR DESCRIPTION
**Pull request by trrenty**
Implementation for the following improvement: https://jira.xwiki.org/browse/CONFLUENCE-166

### CODE REVIEW

**Comment by tmortagne** [comment]
I'm not sure ConfluenceXMLPackage is the right location, since the APIs you have in it are related to exposing what was found during the indexing.

Statistics about what was done during the import sound more like an information you would find in a report, which is not really what ConfluenceXMLPackage is expected to be right now IMO.

Of course, a possibility is to simply log this kind of statistics at the end of ConfluenceInputFilterStream#read. After all, that's exactly what the log is: a report of what happen. But it depends on how you are planning to exploit those data exactly. It's also perfectly possible to reuse the values inserted in the log (as long as you have a way to identify the log entries you are interested in, like a slj Markerf) since the job log contains them, it's not a list of strings.

File: confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceConverterListener.java [hunk 1]
```diff
@@ -45,10 +47,12 @@
 import org.xwiki.contrib.confluence.filter.input.ConfluenceInputContext;
 import org.xwiki.contrib.confluence.filter.input.ConfluenceProperties;
 import org.xwiki.contrib.confluence.filter.input.ConfluenceXMLPackage;
+import org.xwiki.contrib.confluence.filter.internal.ConfluenceFilter;
```
Comment (tmortagne) [inline confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceConverterListener.java:50]: I'm not sure you are really using this import but maybe I missed something.

File: confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/ConfluenceFilter.java [hunk 2]
```diff
@@ -42,5 +44,11 @@ public interface ConfluenceFilter extends WikiFarmFilter, WikiFilter, WikiSpaceF
     WikiAttachmentFilter, WikiClassFilter, WikiClassPropertyFilter, WikiObjectFilter, WikiObjectPropertyFilter,
     UserFilter, GroupFilter, Listener
 {
-
+    /**
+     * Mark a log as the list of macros found during the filtering process. First log parameter should be a list of
+     * strings representing the ids of the macros.
+     *
+     * @since 9.29.6
```
Comment (tmortagne) [inline confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/ConfluenceFilter.java:51]: This change is not a bugfix so it feels more like a 9.30.0

File: confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceConverterListener.java [hunk 3]
```diff
@@ -94,12 +98,41 @@ public class ConfluenceConverterListener extends WrappingListener
...
+    public void setWrappedListener(Listener listener)
+    {
+        wrappingListener.setWrappedListener(listener);
+        super.setWrappedListener(wrappingListener);
+    }
+
+    /**
+     * @param macroIds a set of macro ids that will be updated whenever a new macro event will be called.
+     *
+     * @since 9.29.6
```
Comment (tmortagne) [inline confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceConverterListener.java:129]: Same as above.

File: confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceConverterListener.java (hunk 1 above)
Comment (trrenty) [reply confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceConverterListener.java:50]: They are indeed unused. I expected the build to fail tho..

Comment (tmortagne) [reply confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceConverterListener.java:50]: Now when the file is excluded from checkstyle.

**Pull request merged**